nullaway/src/test/java/com/uber/nullaway/CoreTests.java

```diff
@@ -1154,4 +1155,38 @@ public void enclosingExpressionForNewClassTree() {
             "}")
         .doTest();
   }
+
+  //  @Ignore
+  @Test
+  public void unboxForEachLoop() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Outer.java",
+            "import org.jspecify.annotations.*;",
+            "@NullMarked",
+            "class Test {",
+            "   void f(@Nullable Integer[] array) {",
+            "   // BUG: Diagnostic contains:unboxing of a @Nullable value",
+            "     for (int x : array) {}",
```

Comment:
Please add a negative test, to ensure we don't get a warning when unboxing elements of an array that doesn't have `@Nullable` contents